### PR TITLE
[3.x] Add documentation to GodotSharp

### PR DESCRIPTION
**Pull request by raulsntos**
3.x version of #51079

### CODE REVIEW

File: modules/mono/glue/GodotSharp/GodotSharp/Core/Mathf.cs [hunk 1]
```diff
@@ -42,74 +45,84 @@ public static partial class Mathf
...
-        /// <returns>An angle that would result in the given tangent value. On the range `-Tau/2` to `Tau/2`.</returns>
+        /// <returns>
+        /// An angle that would result in the given tangent value. On the range <c>-Tau/2</c> to <c>Tau/2</c>.
+        /// </returns>
         public static real_t Atan2(real_t y, real_t x)
         {
             return (real_t)Math.Atan2(y, x);
         }
 
-        /// <summary>
```
Comment (aaronfranke) [inline modules/mono/glue/GodotSharp/GodotSharp/Core/Mathf.cs:112]: This doesn't seem right...

**Review by neikeq — APPROVED** [review]

File: modules/mono/glue/GodotSharp/GodotSharp/Core/Mathf.cs (hunk 1 above)
Comment (neikeq) [reply modules/mono/glue/GodotSharp/GodotSharp/Core/Mathf.cs:112]: Yes, this was removed by mistake.

**Review by neikeq — CHANGES_REQUESTED** [review]

File: modules/mono/glue/GodotSharp/GodotSharp/Core/Mathf.cs (hunk 1 above)
Comment (raulsntos) [reply modules/mono/glue/GodotSharp/GodotSharp/Core/Mathf.cs]: Thanks, I missed this when cherry picking. I checked and found a few other mistakes, should be fixed now.

**Review by aaronfranke — APPROVED** [review]

**Comment by aaronfranke** [comment]
@raulsntos Actually I did some backporting of that myself but currently it's mixed in with your doc changes, so I'm waiting on this to be merged before I submit that as a PR.

**Review by neikeq — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**